subprojects/diagnostics/src/main/java/org/gradle/configuration/TaskDetailPrinter.java

```diff
@@ -92,64 +83,48 @@ public void print(StyledTextOutput output) {
             output.println();
             printTaskGroup(output, tasksByType);
 
-            if (multipleClasses) {
+            if (multipleTaskTypes) {
                 output.println();
                 output.println("----------------------");
             }
         }
     }
 
-    private ListMultimap<Class<?>, Task> groupTasksByType(List<Task> tasks) {
-        final Set<Class<?>> taskTypes = new TreeSet<>(Comparator.comparing(Class::getSimpleName));
-        taskTypes.addAll(collect(tasks, this::getDeclaredTaskType));
-
-        ListMultimap<Class<?>, Task> tasksGroupedByType = ArrayListMultimap.create();
-        for (final Class<?> taskType : taskTypes) {
-            tasksGroupedByType.putAll(taskType, filter(tasks, element -> getDeclaredTaskType(element).equals(taskType)));
-        }
-        return tasksGroupedByType;
-    }
-
-    private Class<?> getDeclaredTaskType(Task original) {
```

Comment:
Moved to: https://github.com/gradle/gradle/pull/21419/files#diff-3b5e985c51e6e78724054aee90f48c61b3e0425da0cf96cee6c653f9afcdf3c1R151